components/b2b/AccountHierarchy/AccountHierarchyEditForm/AccountHierarchyEditForm.tsx

```diff
@@ -0,0 +1,138 @@
+      parentAccount: '',
+    },
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+    criteriaMode: 'firstError',
+    resolver: yupResolver(accountHierarchySchema),
+    shouldFocusError: true,
+  })
+
+  useEffect(() => {
+    const hasMultipleAccounts = accounts?.length && accounts.length > 1
+    if (!hasMultipleAccounts) {
+      setValue('parentAccount', accounts?.[0]?.companyOrOrganization as string)
+      setSelectedParentAccount(accounts?.[0])
+    }
+  }, [accounts])
+
+  const onSubmit = () => {
+    if (isLoading || !selectedParentAccount) return
+    setLoading(true)
+    const formValues = getValues()
+    onSave({ parentAccount: selectedParentAccount })
+  }
+
+  const handleParentAccountChange = (name: string, value: string) => {
+    const account: B2BAccount | undefined = accounts?.find(
+      (account: B2BAccount) => account.id === parseInt(value)
+    )
+    setValue('parentAccount', account?.companyOrOrganization as string)
+    setSelectedParentAccount(account)
+  }
+
+  return (
+    <form data-testid="account-hierarchy-form" onSubmit={handleSubmit(onSubmit)}>
+      <Box sx={{ marginBottom: 3 }}>
+        <InputLabel shrink>{t('current-parent-account')}</InputLabel>
+        <Typography>{accountToEdit?.companyOrOrganization}</Typography>
+      </Box>
+
+      <FormControl sx={{ width: '100%' }}>
+        <Controller
+          name="parentAccount"
+          control={control}
+          render={({ field }) => (
+            <KiboSelect
+              sx={{ marginBottom: '20px' }}
+              name="parentAccount"
+              label={t('parent-account')}
+              onChange={handleParentAccountChange}
+              placeholder={t('select-parent-account')}
+              error={!!errors?.parentAccount}
+              helperText={errors?.parentAccount?.message as string}
+              value={selectedParentAccount?.id?.toString() ?? ''}
+            >
+              {accounts?.map((account: B2BAccount) => {
+                return (
+                  <MenuItem key={account?.id} value={`${account?.id}`}>
+                    {account?.companyOrOrganization}
+                  </MenuItem>
+                )
+              })}
+            </KiboSelect>
+          )}
+        />
+
+        <Stack gap={2} sx={{ ...AccountHierarchyFormStyles.buttonStackStyle }}>
+          <Button
+            data-testid="cancel-button"
+            variant="contained"
+            color="secondary"
+            type="reset"
+            onClick={onClose}
+          >
+            {t('cancel')}
+          </Button>
+          <LoadingButton
+            data-testid="submit-button"
+            variant="contained"
+            color="primary"
+            type="submit"
+            loading={isLoading}
+            disabled={isLoading}
+          >
+            {t('update-account')}
+          </LoadingButton>
+        </Stack>
+      </FormControl>
+    </form>
+  )
+}
+
+export default AccountHierarchyEditForm
```

Comment:
As in this form we are not edit entire account but just changing Parent 
We should rename this component some like AccountHierarchyChangeParent

To Edit current account form is missing, add it